nearcore/src/runtime/mod.rs

```diff
@@ -1547,19 +1536,15 @@ impl RuntimeAdapter for NightshadeRuntime {
         let protocol_version = self.get_epoch_protocol_version(epoch_id)?;
         let mut config = self.genesis_config.clone();
         config.protocol_version = protocol_version;
+        // If we were initialised with a custom max_gas_burnt_view (see
+        // ActualRuntimeConfig::new method), keep the value from genesis file.
+        // Since max_gas_burnt_view never changes as a result of protocol
+        // upgrade, we can use it when reporting protocol config.
+        let gas = config.runtime_config.wasm_config.limit_config.max_gas_burnt_view;
```

Comment:
This logic still feels like a smell to me, but I still don't see simple, short improvement. 

I guess, the question I have (going by the naming mostly): why is  `max_gas_burnt_view` is a part of `ProtocolConfig` at all? As far as I understand, view functionality exists outside of the protocol, so perhaps it shouldn't be part of ProtocolConfig?

To ask this in another way: here, we do some logic to get the `max_gas_burnt_view` from the original genesis_config. But what does that number mean? Who uses `max_gas_burnt_view`, as it can be configured separately by each node?

Reply:
Re-reading https://github.com/near/nearcore/issues/3080, it does seem that we want to remove the field from GenesisConfig (or at least make it unused). @bowenwang1996 , if that's roughly correct, could you create an issue for that refactor? I am not feeling good about having two sources of truth for `max_gas_burnt_view` -- it won't cause bugs, but it is tech debt.

Reply:
> To ask this in another way: here, we do some logic to get the max_gas_burnt_view from the original genesis_config. But what does that number mean?

It means nothing at all. ;) At this point it’s essentially the default value for the limit which is used if client doesn’t override it. In fact we *probably* could return the modified value here, but having different client return different ‘protocol config’ does feel weird as well. (For some more context, this value is returned by `EXPERIMENTAL_protocol_config` RPC).

I think part of the issue is that this value is part of the genesis hash, though I might be misunderstandingn how hashing works.

Reply:
@matklad I created https://github.com/near/nearcore/issues/4447 and @posvyatokum is working on that